### PR TITLE
network, Istio: Fix outbound test

### PR DESCRIPTION
**Pull request by rhrazdil**
**What this PR does / why we need it**:

The negative test checking that Envoy refuses to
connect to external service was missing .Should(Succeed())
so when we changed from using "kubevirt.io" as an external
service to another cluster vmi, the test was broken without
us knowing about it.

This commit properly registers the server vmi (there are two VMIs, server VMI without proxy 
sidecar and the tested client VMI which does have the proxy sidecar) as a service by defining
VirtualService, destinationRule and Gateway.

The vmi HTTP server is now accessed via Istio ingressgateway, and when 
Sidecar disallowing all external services is created, the Envoy proxy properly
refuses to forward the request.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/network/vmi_istio.go [hunk 1]
```diff
@@ -423,3 +452,97 @@ func generateRegistryOnlySidecar() *unstructured.Unstructured {
 		},
 	}
 }
+
+func generateVirtualService() *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "networking.istio.io/v1beta1",
```
Comment (alonSadan) [inline tests/network/vmi_istio.go:459]: perhaps define "networking.istio.io/v1beta1" as const?

File: tests/network/vmi_istio.go [hunk 2]
```diff
@@ -423,3 +452,97 @@ func generateRegistryOnlySidecar() *unstructured.Unstructured {
...
+			"kind":       "VirtualService",
+			"metadata": map[string]interface{}{
+				"name": "vmi-server-vs",
+			},
+			"spec": map[string]interface{}{
+				"gateways": []string{
+					"vmi-server-gw",
```
Comment (alonSadan) [inline tests/network/vmi_istio.go:466]: ditto for "vmi-server-gw"

File: tests/network/vmi_istio.go [hunk 3]
```diff
@@ -261,25 +267,54 @@ var _ = SIGDescribe("[Serial] Istio", func() {
...
 				Expect(err).ShouldNot(HaveOccurred())
-				serverVMIAddress = serverVMI.Status.Interfaces[0].IP
+
+				By("Creating Istio Gateway")
+				gatewaysRes := schema.GroupVersionResource{Group: "networking.istio.io", Version: "v1beta1", Resource: "gateways"}
+				gateway := generateGateway()
+				_, err = virtClient.DynamicClient().Resource(gatewaysRes).Namespace(util.NamespaceTestDefault).Create(context.TODO(), gateway, metav1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+
+				By("Getting back Istio ingressgateway IP")
```
Comment (alonSadan) [inline tests/network/vmi_istio.go:306]: nit: why "getting back" and not just "getting" ?

File: tests/network/vmi_istio.go [hunk 4]
```diff
@@ -261,25 +267,54 @@ var _ = SIGDescribe("[Serial] Istio", func() {
...
 				Expect(err).ToNot(HaveOccurred())
-				Expect(console.LoginToCirros(serverVMI)).To(Succeed())
 
+				serverVMIService := newService(
+					"vmi-server",
+					map[string]string{"app": vmiServerAppSelector},
+					[]corev1.ServicePort{{Port: vmiServerTestPort}},
+				)
+				_, err = virtClient.CoreV1().Services(util.NamespaceTestDefault).Create(context.Background(), serverVMIService, metav1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
```
Comment (alonSadan) [inline tests/network/vmi_istio.go:282]: nit: ShouldNot -> ToNot. 
sounds better I think

File: tests/network/vmi_istio.go [hunk 5]
```diff
@@ -261,25 +267,54 @@ var _ = SIGDescribe("[Serial] Istio", func() {
...
+				Expect(console.LoginToCirros(serverVMI)).To(Succeed())
 				By("Starting HTTP Server")
-				tests.StartHTTPServer(serverVMI, testPort)
+				tests.StartHTTPServer(serverVMI, vmiServerTestPort)
+
+				By("Creating Istio VirtualService")
+				virtualServicesRes := schema.GroupVersionResource{Group: "networking.istio.io", Version: "v1beta1", Resource: "virtualservices"}
+				virtualService := generateVirtualService()
+				_, err = virtClient.DynamicClient().Resource(virtualServicesRes).Namespace(util.NamespaceTestDefault).Create(context.TODO(), virtualService, metav1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
```
Comment (alonSadan) [inline tests/network/vmi_istio.go:292]: ditto

File: tests/network/vmi_istio.go [hunk 6]
```diff
@@ -261,25 +267,54 @@ var _ = SIGDescribe("[Serial] Istio", func() {
...
+				destinationRule := generateDestinationRule()
+				_, err = virtClient.DynamicClient().Resource(destinationRulesRes).Namespace(util.NamespaceTestDefault).Create(context.TODO(), destinationRule, metav1.CreateOptions{})
 				Expect(err).ShouldNot(HaveOccurred())
-				serverVMIAddress = serverVMI.Status.Interfaces[0].IP
+
+				By("Creating Istio Gateway")
+				gatewaysRes := schema.GroupVersionResource{Group: "networking.istio.io", Version: "v1beta1", Resource: "gateways"}
+				gateway := generateGateway()
+				_, err = virtClient.DynamicClient().Resource(gatewaysRes).Namespace(util.NamespaceTestDefault).Create(context.TODO(), gateway, metav1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
```
Comment (alonSadan) [inline tests/network/vmi_istio.go:304]: ditto

**Review by alonSadan — COMMENTED** [review]
Thank you Radim:)
just some nits

File: tests/network/vmi_istio.go (hunk 3 above)
Comment (rhrazdil) [reply tests/network/vmi_istio.go]: I don't know either, Done

File: tests/network/vmi_istio.go (hunk 4 above)
Comment (rhrazdil) [reply tests/network/vmi_istio.go]: Done

File: tests/network/vmi_istio.go (hunk 1 above)
Comment (rhrazdil) [reply tests/network/vmi_istio.go]: Done

File: tests/network/vmi_istio.go (hunk 2 above)
Comment (rhrazdil) [reply tests/network/vmi_istio.go]: Done

**Review by alonSadan — APPROVED** [review]
Thank you Radim!

**Comment by AlonaKaplan** [comment]
/approve
Thanks!

**Pull request merged**